CMakeLists.txt

```diff
@@ -304,33 +322,39 @@ if(APPLE)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libc++")
 endif()
 
-if (NOT MSVC)
-    add_cxx_flag_if_supported("-Wall")
-    add_cxx_flag_if_supported("-Wextra")
-    add_cxx_flag_if_supported("-Wunused")
-    add_cxx_flag_if_supported("-pedantic")
-    add_cxx_flag_if_supported("-Wsign-compare")
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
```

Comment:
Given how MSVC handles PIC (i.e., it doesn't), I think we should move this into the `if (NOT MSVC)` block, unless you have any objections?

Reply:
This option does nothing automatically if MSVC is used. If you wish, I can move it under `if (NOT MSVC)`

Reply:
Agreed, but it is a "nicer" change if we exchange `-fPIC` for `CMAKE_POSITION_INDEPENDENT_CODE`!

Reply:
No problem, will move!

Reply:
Done!